Code/Tools/FBuild/FBuildCore/Helpers/VSProjectGenerator.cpp

```diff
@@ -137,17 +224,51 @@ const AString & VSProjectGenerator::GenerateVCXProj( const AString & projectFile
                     break;
                 }
             }
+
+            const bool   isHeader = IsFilenameAHeaderFile( *fIt );
+            const char * includeTypeString = isHeader ? "ClInclude" : "CustomBuild";
+            Write( "    <%s Include=\"%s\">\n", includeTypeString, fileName );
             if ( fileType )
             {
-                Write( "    <CustomBuild Include=\"%s\">\n", fileName );
                 Write( "        <FileType>%s</FileType>\n", fileType );
-                Write( "    </CustomBuild>\n" );
             }
-            else
+
+            if ( writeCustomBuildCommands && !isHeader )
             {
-                Write( "    <CustomBuild Include=\"%s\" />\n", fileName );
+                // Build Command [Conditional]
+                Write( "        <Message>Compiling %%(Identity)</Message>\n" );
+                const VSProjectConfig * const cEnd = configs.End( );
+                for ( const VSProjectConfig * cIt = configs.Begin( ); cIt != cEnd; ++cIt )
+                {
```

Comment:
I'm a little confused about what this custom build command is for. In what situation is this used?

Reply:
This is for `CTRL+F5` iterative builds, where you can select a single source file to build via visual studio. This uses the build command directly to build that source file at the command line (no linking).

Reply:
While this might work (in that the file will compile) it's not really safe - i.e. you can't mix and match compiling file directly outside of FASTBuild and then using FASTBuild.  Best case it will rebuild when you switch back to a proper FASTBuild build, but worst case you will get weird failures.  I have also had problems with Visual Studio project conditionals (not working properly when switching between build configurations, only being re-evaluated when quitting VS and reloading the sln).

I'd like to explore getting Ctrl+F5 working, but if we could decouple this from the Android support that would be good.